src/IssueViz.Security.UnitTests/Taint/TaintList/TaintIssuesControlViewModelTests.cs

```diff
@@ -232,6 +233,69 @@ public void Dispose_UnregisterFromSelectionChangedEvent()
             selectionService.VerifyRemove(x => x.SelectedIssueChanged -= It.IsAny<EventHandler>(), Times.Once());
         }
 
+        [TestMethod]
+        public void ShowVisualizationPaneCommand_CanExecute_NullParameter_False()
+        {
+            var menuCommandService = new Mock<IMenuCommandService>();
+            var testSubject = CreateTestSubject(menuCommandService: menuCommandService.Object);
+
+            VerifyCommandExecution(testSubject.ShowVisualizationPaneCommand, null, false);
+
+            menuCommandService.VerifyNoOtherCalls();
+        }
+
+        [TestMethod]
+        public void ShowVisualizationPaneCommand_CanExecute_ParameterIsNotTaintViewModel_False()
+        {
+            var menuCommandService = new Mock<IMenuCommandService>();
+            var testSubject = CreateTestSubject(menuCommandService: menuCommandService.Object);
+
+            VerifyCommandExecution(testSubject.ShowVisualizationPaneCommand, "something", false);
+
+            menuCommandService.VerifyNoOtherCalls();
+        }
+
+        [TestMethod]
+        public void ShowVisualizationPaneCommand_CanExecute_ParameterIsTaintViewModel_True()
+        {
+            var menuCommandService = new Mock<IMenuCommandService>();
+            var testSubject = CreateTestSubject(menuCommandService: menuCommandService.Object);
+
+            VerifyCommandExecution(testSubject.ShowVisualizationPaneCommand, Mock.Of<ITaintIssueViewModel>(), true);
+
+            menuCommandService.VerifyNoOtherCalls();
+        }
+
+        [TestMethod]
+        public void ShowVisualizationPaneCommand_Execute_LocationNavigated()
+        {
+            var menuCommandService = new Mock<IMenuCommandService>();
+
+            var viewModel = new Mock<ITaintIssueViewModel>();
+
+            var testSubject = CreateTestSubject(menuCommandService: menuCommandService.Object);
+            testSubject.ShowVisualizationPaneCommand.Execute(viewModel.Object);
+
+            menuCommandService.Verify(x => x.GlobalInvoke(It.IsAny<CommandID>()), Times.Once);
```

Comment:
Is there are reason not to check the expected command is being called?

Reply:
Because it's a reference type i didn't think it would work. But after the comment I've looked at the de-compiled code and saw equals method was overridden

Reply:
An alternative approach in that case would have been to capture the value that was passed and assert against the individual fields instead.